authconfig/linkedin.ts

```diff
@@ -18,23 +18,48 @@ passport.use(
         // represent the logged-in user. In a typical application, you would want
         // to associate the LinkedIn account with a user record in your database,
         // and return that user instead.
-        const findUser = await User.findBy({ email: profile.emails[0].value });
 
+        // Checks for the user by the linkedin profile id
+        const findUser = await User.findBy({ auth_id: profile.id });
+
+        // if no user is found attempt to create a new user
         if (!findUser) {
           const pw = await hash(profile.displayName, 12);
 
+          // IF a user exists with the same email, return that user
```

Comment:
This is a good change. As I understand it, it would be checking to see if a user already exists and if so it would be saving the provider to their user object so even if they clicked the wrong button it would still return them?

Reply:
Yea, there are some things I want to change about it tonight now that I look at it, and still something with the usernames to discuss. There is still a chance different strategies could use a different email but the same profile name.

That causes an error with usernames not being unique